src/wallet.js

```diff
@@ -224,21 +224,65 @@ const wallet = {
 
       storage.setItem('wallet:data', dataJson);
 
-      walletApi.getAddressHistory(addresses, (response) => {
+      this.getHistory(addresses).then((history) => {
         const data = this.getWalletData();
         // Update historyTransactions with new one
         const historyTransactions = 'historyTransactions' in data ? data['historyTransactions'] : {};
         const allTokens = 'allTokens' in data ? data['allTokens'] : [];
-        const result = this.updateHistoryData(historyTransactions, allTokens, response.history, resolve, data, reject);
+        const result = this.updateHistoryData(historyTransactions, allTokens, history, resolve, data, reject);
         WebSocketHandler.emit('addresses_loaded', result);
-      }).catch((e) => {
-        // Error in request
+      }, (e) => {
         reject(e);
       });
+
     });
     return promise;
   },
 
+  /**
+   * Asynchronous method to get history of an array of transactions
+   * Since this API is paginated, we enter a loop getting all data and return only after all requests are finished
+   *
+   * @param {Array} addresses Array of addresses (string) to get history
+   *
+   * @return {Promise} Promise that resolves when all addresses history requests finish
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  async getHistory(addresses) {
```

Comment:
Maybe rename to `getTxHistory` or `getAddressHistoryNewApi`?

Reply:
Renamed to `getTxHistory`